### DISCUSSION TITLE
dynamic.py, __str__ and unicode

### DISCUSSION THREAD

**Discussion by ghost** [mhammond/pywin32#2515] [General]
I had a problem where a 3rd party com object was unexpectedly outputting unicode data which would making my program's str\(\) blow up on that property. The only solution I could find was to modify the dynamic.py program.

dynamic.py: Line 198; Insert
except UnicodeEncodeError:
return self.\_\_call\_\_\(\).encode\('utf-7'\)

Is their a better solution? If not please add this or something like it. Thanks\!

Reported by: closedbox

Original Ticket: [pywin32/bugs/439](https://sourceforge.net/p/pywin32/bugs/439)

**Comment by ghost** [comment]
woops that line 189 not 198

Original comment by: closedbox

**Comment by Avasam** [comment]
Python 2 to 3 migration, bytes vs str handling
